src/pybind/mgr/orchestrator/module.py

```diff
@@ -1188,6 +1188,27 @@ def _apply_rgw(self,
                 out = to_format(data, format, many=True, cls=None)
         return HandleCommandResult(stdout=out)
 
+    @_cli_write_command(
+        'orch apply ha_rgw',
+        'Update the number of ha_rgw instances')
+    def _apply_ha_rgw(self,
+                       inbuf: Optional[str] = None) -> HandleCommandResult:
+
+        usage = """Usage:
+  ceph orch apply ha_rgw -i <yaml spec>
+        """
+        if inbuf:
+            s = yaml.safe_load(inbuf)
+            spec = json_to_generic_spec(s)
```

Comment:
Since we have a bunch of custom parameters for this service, will generating a generic spec here suffice or do we need to create a custom service spec class for this service (similar to RGWSpec) and generate that instead?

Reply:
discussed in standup